Parsers/ASimAuthentication/ProductParsers/AuthenticationM365D.yaml

```diff
@@ -0,0 +1,131 @@
+- Title: ASIM
+  Link: https:/aka.ms/AzSentinelNormalization
+Description: |
+      This Query Parser maps M365 Defender Device File Events (DeviceFileEvents) to the Azure Sentinel Information Model authenticaion schema.
+ParserName: vimAuthenticationM365Defender
+ParserQuery: |
+  let FaliureReason=datatable(EventOriginalResultDetails:string, EventResultDetails:string)[
+    'InvalidUserNameOrPassword','No such user or password' 
+    // What other will we find here?
+    ];
+  let AuthM365D=(){
+    DeviceLogonEvents 
+    // ---   Already mapped
+    // // TimeGenerated - exists
+    //
+    | project-rename 
+       EventOriginalResultDetails=FailureReason 
+       , EventSubType=LogonType
+    | extend 
+    //
+       // ----  Untouched
+       TimeGenerated
+    // 
+      // ----  Event
+       , EventCount=int(1)
+       , EventStartTime=TimeGenerated
+       , EventEndTime=TimeGenerated
+       , EventOriginalType = EventSubType // Previously:LogonType
+       , EventProduct='M365 Defender for EndPoint'
+      ,  EventResult = case(ActionType =='LogonSuccess', 'Success'
+                            , ActionType=='LogonFailed', 'Failure'
+                            , ActionType=='LogonAttempted', 'NA' //// Not sure what this is
+                            , 'NA')
+      , EventSchemaVersion='0.1.0'
+      , EventType='Logon'
+      , EventVendor ='Microsoft'
+    //    -   No Data
+    //  EventOriginalUid 
+    //  EventProductVersion 
+    //  EventReportUrl
+    //  EventMessage 
+    //  
+    //  
+    //  ----   Target and Actor Users
+     | project-rename 
+         TargetUserId=AccountSid
+       , ActorUserId =InitiatingProcessAccountSid
+       , ActorUserUpn=InitiatingProcessAccountUpn
+       , ActorUserObjectId=InitiatingProcessAccountObjectId
+      | extend 
+         TargetUserIdType ='UserSid'
+       , TargetUsername = strcat(AccountDomain,'\\',AccountName)
+       , TargetUsernameType='Windows'
+    //    -  No Data
+      // TargetUserType 
+    // 
+    // ActingAppName 
+    // ActingAppType 
+    // ActorSessionId 
+      , ActorUserIdType='SID'
+    // ActorUserType 
+      , ActorUsername=coalesce(ActorUserUpn, strcat(InitiatingProcessAccountDomain,'\\',InitiatingProcessAccountName)) // InitiatingProcessAccountName
+      , ActorUsernameType=case(isnotempty( ActorUserUpn), 'Upn/Email'
+                                , isnotempty(InitiatingProcessAccountDomain), 'Windows'
+                                , 'Simple')
+     //, AdditionalFields:already exists
+      | project-rename 
+        DvcHostname=DeviceName
+      , LogonProtocol=Protocol
+      , TargetDvcId=DeviceId
+      , TargetDvcIpAddr=RemoteIP
+      , OriginalEventUid=ReportId
+    // No Data
+    // DvcIpAddr 
+    // HttpUserAgent 
+    // LogonMethod 
+    //
+      , SrcDvcHostname=RemoteDeviceName 
+      // SrcDvcHostnameType == Requires calculation. not sure it is worth it
+    // SrcDvcId 
+    // SrcDvcIpAddr 
```

Comment:
SrcIpAddr should be RemoteIP. SrcHostname shuold be RemoteDeviceName, SrcPort = RemotePort

Reply:
@oshezaf 
https://docs.microsoft.com/en-us/azure/sentinel/normalization#the-device-entity

RemoteIP | string | IP address that was being connected to [so it is target, in our terms]

similarly, RemotePort.

RemoteDeviceName is the Name of the machine that performed a remote operation

Reply:
verifying that this is not a mistake in documentation